src/ray/raylet/task_dependency_manager.cc

```diff
@@ -85,6 +85,7 @@ std::vector<TaskID> TaskDependencyManager::HandleObjectLocal(
     auto object_entry = creating_task_entry->second.find(object_id);
     if (object_entry != creating_task_entry->second.end()) {
       for (auto &dependent_task_id : object_entry->second) {
+        UnsubscribeWaitDependency(dependent_task_id, object_id);
         auto &task_entry = task_dependencies_[dependent_task_id];
         task_entry.num_missing_dependencies--;
```

Comment:
I think we should only consider `ray.get` dependencies when decrementing `num_missing_dependencies`. Probably use the return value of `UnsubscribeWaitDependency` to check whether we should decrement.